scripting/pugsetup/mapvote.sp

```diff
@@ -4,8 +4,14 @@
  * Map voting functions
  */
 public void CreateMapVote() {
+  if (g_ExcludedMaps.IntValue > 0 && g_MapList.Length > g_PastMaps.Length) {
+    GetMapVotePool();
+  } else {
+    g_MapVotePool = g_MapList;
```

Comment:
This line scares me a little since it could result in the changes to g_MapVotePool affecting the value of g_MapList since this is essentially copying the reference to an ArrayList. 

Since GetMapVotePool works just by copying data from g_MapList into g_MapVotePool, I think this should do the same. You could change GetMapVotePool to take a bool excludeRecentMaps variable and reuse that function in both cases then.